bundles/config/org.eclipse.smarthome.config.core/src/main/java/org/eclipse/smarthome/config/core/ConfigDescriptionParameterGroup.java

```diff
@@ -25,30 +28,26 @@
  * @author Chris Jackson - Initial Contribution
  *
  */
+@NonNullByDefault
 public class ConfigDescriptionParameterGroup {
 
-    private String name;
-    private String context;
-    private boolean advanced;
-    private String label;
-    private String description;
+    private final String name;
+    private final @Nullable String context;
+    private final boolean advanced;
+    private final String label;
+    private final String description;
 
     /**
      * Create a Parameter Group. A group is used by the user interface to display groups
      * of parameters together.
      *
-     * @param name
-     *            the name, used to link the group, to the parameter
-     * @param context
-     *            a context string. Can be used to provide some context to the group
-     * @param advanced
-     *            a flag that is set to true if this group contains advanced settings
-     * @param label
-     *            the human readable group label
-     * @param description
-     *            a description that can be provided to the user
+     * @param name the name, used to link the group, to the parameter
+     * @param context a context string. Can be used to provide some context to the group
+     * @param advanced a flag that is set to true if this group contains advanced settings
+     * @param label the human readable group label
+     * @param description a description that can be provided to the user
      */
-    public ConfigDescriptionParameterGroup(String name, String context, Boolean advanced, String label,
+    public ConfigDescriptionParameterGroup(String name, @Nullable String context, Boolean advanced, String label,

```

Comment:
Why is "advanced" a Boolean and not a boolean here? If we require NonNull, making it a boolean might be a worthwhile change here.

Reply:
I asked me the same...

In the constructor the `Boolean` object unboxed without checks to the primitive `boolean` type.
So I assume the "non null by default" usage is correct for the argument.

I don't want to change the binary compatibility, so I don't touched the argument type.

Reply:
Ok, you are probably right that it is not worth to break binary compatibility for this.